stock_picking_low_priority/tests/test_stock_picking_low_priority.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields
+from odoo.tests import Form, TransactionCase
+
+
+class TestStockPickingLowPriority(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.PickingObj = cls.env["stock.picking"]
+        cls.MoveObj = cls.env["stock.move"]
+        cls.supplier_location = cls.env.ref("stock.stock_location_suppliers").id
+        cls.stock_location = cls.env.ref("stock.stock_location_stock").id
+        cls.customer_location = cls.env.ref("stock.stock_location_customers").id
+        cls.picking_type_out = cls.env.ref("stock.picking_type_out").id
+        cls.picking_type_in = cls.env.ref("stock.picking_type_in").id
+        cls.uom_unit = cls.env.ref("uom.product_uom_unit")
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Product Test",
+                "uom_id": cls.uom_unit.id,
+                "type": "product",
+            }
+        )
+
+    def test_selection_insertion(self):
+        """
+        Test that the 'Not urgent' priority has been inserted in first position of the
+        selection
+        """
+        priority_selection = self.PickingObj.fields_get()["priority"]["selection"]
+        self.assertEqual(priority_selection[0], ("-1", "Not urgent"))
+
+    def test_assign_qty_to_first_move(self):
+        """
+        Suppose two out picking waiting for an available quantity. When receiving such
+        a quantity, the latter should be assign to the picking with the
+        highest priority.
+        """
+
+        def create_picking(picking_type, from_loc, to_loc, sequence=10, delay=0):
```

Comment:
@glitchov You must avoid to define nested method. It make the code difficult to maintain and extend.